src/PIL/TgaImagePlugin.py

```diff
@@ -166,11 +174,12 @@ def _save(im, fp, filename):
     else:
         flags = 0
 
-    orientation = im.info.get("orientation", -1)
+    orientation = im.encoderinfo.get("orientation",
+                                     im.info.get("orientation", -1))
     if orientation > 0:
         flags = flags | 0x20
 
-    fp.write(b"\000" +
+    fp.write(o8(idlen) +
```

Comment:
There is a small problem: o8 is `bytes((i & 255,))`, which will not raise an exception like `struct.pack()`, making it possible for `fp.write()` below to write `id_section` bigger than 255 bytes, resulting in a broken image.

Reply:
Okay, I've added a commit to trim `id_section`.